src/libstd/thread.rs

```diff
@@ -275,6 +275,10 @@ impl Builder {
             unsafe {
                 stack::record_os_managed_stack_bounds(my_stack_bottom, my_stack_top);
             }
+            match their_thread.name() {
+                Some(thename) => unsafe { imp::set_name(thename.as_slice()); },
+                None => {}
+            }
```

Comment:
Could this be folded into `thread_info::set` to benefit the main thread as well?